Mathlib/Data/List/Iterate.lean

```diff
@@ -17,6 +17,14 @@ variable {α : Type*}
 
 namespace List
 
+theorem take_range (m n : ℕ) : take m (range n) = range (min m n) := by
+  apply List.ext_get
+  · simp
+  · simp only [length_take, length_range, lt_min_iff, get_range]
+    intro i h _
+    rw [← get_take _ _ h.1, get_range]
+    simpa using h.2
```

Comment:
```suggestion
theorem take_range (m n : ℕ) : take m (range n) = range (min m n) := by
  apply List.ext_get <;> simp (config := { contextual := true }) [← get_take]
```
also, I think this theorem should be moved to `Mathlib.Data.List.Range`.

Reply:
I think `lt_min_iff` is not available in that file

Reply:
Use `Nat.lt_min` in `Init.Data.Nat.MinMax` instead.

Reply:
Nice, thanks!